### DISCUSSION TITLE
How to deploy node.js express app?

### DISCUSSION THREAD

**Discussion by Dan-Veinberg** [vercel/community#4045] [Feature Question]
### Question

How do I deploy a node.js express.js app? I deployed something but my routes are not working. I am getting a 404 not found error. In addition my app needs to spawn child python process to run a short python script. I am not very familiar with vercel. It is my first time using it. Would vercel allow me to run python scripts even though it is a node.js app? 

To summarize my questions

1. How  to properly deploy node.js express.js app?
2. Is it possible to run python scripts? 
3. Is it possible to use python 3.11+ 
4. Is it possible to manage python packages with pip?
5. How to see logs of node.js app?

**Comment by amyegan** [comment]
Hey @Dan-Veinberg! It is possible to have both Node.js and Python serverless functions. You can find more info on the following pages:

- [Using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel)
- [Using the Node.js Runtime with Serverless Functions](https://vercel.com/docs/functions/serverless-functions/runtimes/node-js#)
- [Using the Python Runtime with Serverless Functions](https://vercel.com/docs/functions/serverless-functions/runtimes/python#)
- [Runtime Logs](https://vercel.com/docs/observability/runtime-logs#)
- [Working with Log Drains](https://vercel.com/docs/observability/log-drains-overview/log-drains#)
